src/lib.rs

```diff
@@ -269,43 +284,55 @@ pub struct Watch {
 }
 
 impl Watch {
-    pub fn watch_path(&mut self, path: impl AsRef<Path>) {
-        self.watch_paths.push(path.as_ref().to_path_buf())
+    pub fn watch_path(&mut self, path: impl AsRef<Path>) -> &mut Self {
+        self.watch_paths.push(path.as_ref().to_path_buf());
+        self
     }
 
-    pub fn watch_paths(&mut self, paths: impl IntoIterator<Item = impl AsRef<Path>>) {
+    pub fn watch_paths(&mut self, paths: impl IntoIterator<Item = impl AsRef<Path>>) -> &mut Self {
         for path in paths {
-            self.watch_path(path)
+            self.watch_path(path);
         }
+        self
     }
 
-    pub fn exclude_path(&mut self, path: impl AsRef<Path>) {
-        self.exclude_paths.push(path.as_ref().to_path_buf())
+    pub fn exclude_path(&mut self, path: impl AsRef<Path>) -> &mut Self {
+        self.exclude_paths.push(path.as_ref().to_path_buf());
+        self
     }
 
-    pub fn exclude_paths(&mut self, paths: impl IntoIterator<Item = impl AsRef<Path>>) {
+    pub fn exclude_paths(
+        &mut self,
+        paths: impl IntoIterator<Item = impl AsRef<Path>>,
+    ) -> &mut Self {
         for path in paths {
-            self.exclude_path(path)
+            self.exclude_path(path);
         }
+        self
     }
 
-    pub fn exclude_workspace_path(&mut self, path: impl AsRef<Path>) {
+    pub fn exclude_workspace_path(&mut self, path: impl AsRef<Path>) -> &mut Self {
         self.workspace_exclude_paths
-            .push(metadata().workspace_root.as_std_path().join(path))
+            .push(path.as_ref().to_path_buf());
+        self
     }
 
-    pub fn exclude_workspace_paths(&mut self, paths: impl IntoIterator<Item = impl AsRef<Path>>) {
+    pub fn exclude_workspace_paths(
+        &mut self,
+        paths: impl IntoIterator<Item = impl AsRef<Path>>,
+    ) -> &mut Self {
         for path in paths {
-            self.exclude_workspace_path(path)
+            self.exclude_workspace_path(path);
         }
+        self
     }
 
     fn is_excluded_path(&self, path: &Path) -> bool {
         self.exclude_paths.iter().any(|x| path.starts_with(x))
             || self
                 .workspace_exclude_paths
                 .iter()
-                .any(|x| path.starts_with(x))
+                .any(|x| path.starts_with(metadata().workspace_root.as_std_path().join(x)))
```

Comment:
This algorithm could be improved to give less work to the machine. But it works so... 🤷‍♀️ it REALLY doesn't matter.

On the other hand it is good for you to think about problems differently so please give it a thought.

To improve it you need to:
1. check that `path` has `workspace_root` as prefix, if not => there is no need to check the `workspace_exclude_paths` at all
2. then you strip the workspace root prefix from path (1 allocation because happens only once)
3. then you check every path in `workspace_exclude_paths` against this